core/trino-spi/src/main/java/io/trino/spi/security/Identity.java

```diff
@@ -130,13 +130,15 @@ public boolean equals(Object o)
             return false;
         }
         Identity identity = (Identity) o;
-        return Objects.equals(user, identity.user);
+        return Objects.equals(user, identity.user)
```

Comment:
I would extract this commit to separate pull request and also I would love to add a test to `TestAccessControl` to expose the issue it may cause.

I think we could consider skipping groups as they are not used access control.

I mean separate pull request is needed as this changes the semantic so it is dense.

Reply:
> I would extract this commit to separate pull request and also I would love to add a test to `TestAccessControl` to expose the issue it may cause.

I'm not sure if this can be triggered currently. I know of only one place (so far) where this may cause issues and it would require somehow requiring two access checks with the same use but with different sets of enabled roles. This can only happen when views are involved but, before the change in this PR, whenever a view changes the identity, it always changes the user.

Reply:
I extracted #14130, but in the meantime I discovered some more questions :) (They're in a comment in that PR)